DataFormats/HGCalReco/interface/TICLCandidate.h

```diff
@@ -20,13 +20,30 @@ class TICLCandidate : public reco::LeafCandidate {
   TICLCandidate() : LeafCandidate(), time_(0.f), timeError_(-1.f), rawEnergy_(0.f), idProbabilities_{} {}
 
   TICLCandidate(const edm::Ptr<ticl::Trackster>& trackster)
-      : LeafCandidate(), time_(0.f), timeError_(-1.f), rawEnergy_(0.f), tracksters_({trackster}), idProbabilities_{} {}
+      : LeafCandidate(),
+        time_(trackster->time()),
+        timeError_(trackster->timeError()),
+        rawEnergy_(0.f),
+        tracksters_({trackster}),
+        idProbabilities_{} {}
 
   inline float time() const { return time_; }
   inline float timeError() const { return timeError_; }
 
   void setTime(float time) { time_ = time; };
   void setTimeError(float timeError) { timeError_ = timeError; }
+  void computeTime() {
+    auto time = 0.;
+    auto timeErr = 0.;
+    for (const auto& tr : tracksters_) {
+      time += tr->time() / pow(tr->timeError(), 2);
+      timeErr += pow(tr->timeError(), -2);
+    }
+    timeErr = sqrt(1 / timeErr);
+
+    setTime(time * pow(timeErr, 2));
+    setTimeError(timeErr);
+  }
```

Comment:
This function is non-protected against all possible division by zero that could happen.
Are we protected against the default-value case, i.e., in the cases in which the `Trackster`s have no timing information?
There is also no need to call member functions within this, right? You could assign directly the values to `time_` and `timeError_`.
There is also an empty constructor of a`TICLCandidate`: calling this method on objects built with that, will result in `nan`s. That's the case, for example, for the `trackster`s promoted from unused `track-seeds`: by definition, they won't have any associated trackster.

Reply:
Also, please do not define the method that computes the time inside DataFormats: rather compute it in the  TrackstersMergeProducer (also following the [suggestion](https://github.com/cms-sw/cmssw/pull/32046#discussion_r519917688) of @rovere), and propagate the computed values to the TICLCandidate with the `setTime` and `setTimeError` methods.

Reply:
Ok, I moved the method from `DataFormats` to `TrackstersMergeProducer` and added the protections against all possible divisions by zero:
https://github.com/cms-sw/cmssw/pull/32046/commits/28193e02be3d0cc905ba13672e9d34a1566fff69?file-filters%5B%5D=.cc#diff-b7e265e02507bf39d28c83598cf1508bc54f9b23c4787ce7ba8b3fca276d4caa